### ISSUE TITLE
keras ImageDataGenerator  issue

### ISSUE THREAD

**Issue by DNJ-JINI** [keras-team/keras#17588]
I am trying to do runtime augmentation using ImageDataGenerator for my segmentation task( 3 objects).
The mask image values are 0,128 and 255 corresponding to each object. but when the image generator is applied the obtained mask value turns out to be 0 and 255 all the 128 values are converted to 255. I am attaching the input image,augemnted mask image and the code to reproduce the issue.
![img_orig](https://user-images.githubusercontent.com/67704244/220603093-fb6aa2a8-f103-4fe0-a401-c447952f6cf9.png)
![img1_aug](https://user-images.githubusercontent.com/67704244/220603151-1b7f5e11-30b1-4c67-8528-98a89f6e2dbc.png)

But this issue is  not occurring for all images. Image for this case is also attached for reference

![img_2_orig](https://user-images.githubusercontent.com/67704244/220603858-c2d621f3-6365-4ac4-bd2e-9bc5b67454f2.png)
![img2_aug](https://user-images.githubusercontent.com/67704244/220604039-c13acc11-f9ab-4a0b-96d4-d67875784a8b.png)

code used:

from keras.preprocessing.image import ImageDataGenerator
from keras.preprocessing.image import img_to_array
from keras.preprocessing.image import load_img

import os
import numpy as np

IMAGE_PATH = "./images"
OUTPUT_DIRECTORY = "./aug_images/"
for i in os.listdir(IMAGE_PATH): 
	image = load_img(os.path.join(IMAGE_PATH ,i),color_mode="grayscale")
	image = img_to_array(image)
	#print(image)
	image = np.expand_dims(image, axis=0) 
	#image  = (image).astype(np.uint8)

	datagen = ImageDataGenerator(brightness_range=[0.4,1.5],zoom_range=0.3)
	PREFIX = 'aug'
	imGen = datagen.flow(image, batch_size=1, save_to_dir = OUTPUT_DIRECTORY, 
                    save_prefix=PREFIX, save_format='png')
	batch = imGen.next()

Any help on this will be greatly appreciated.

**Comment by tilakrayal** [comment]
@DNJ-JINI,
I was facing the indentation error while executing the code. Kindly find the gist of it here and provide the code in the reproducible way or the colab gist which helps to debug the issue. Thank you!
